### PR TITLE
Expose function to add entropy tables to pre-built dictionary.

### PR DESCRIPTION
**Pull request by ot**
In some cases a custom dictionary building algorithm tailored for a specific
input can be more effective than the one produced by `ZDICT_trainFromBuffer`,
but with the current API it's not possible encode the entropy tables into the
custom-built dictionary.

This commit extracts the logic to add entropy tables to a dictionary from
`ZDICT_trainFromBuffer` and exposes it as a function
`ZDICT_addEntropyTablesFromBuffer`.

### CODE REVIEW

**Comment by Cyan4973** [comment]
Great idea Giuseppe !

Also, note that the project doesn't take PR directly into master.

It should go first through "dev" branch, or even its own feature branch.
Note that the currently active feature branch is dev070.
It features some little changes within `zdict` which will lightly affect your patch (`dictID` header field).

File: lib/dictBuilder/zdict.c [hunk 1]
```diff
@@ -810,6 +810,31 @@ static size_t ZDICT_fastSampling(void* dictBuffer, size_t dictSize,
...
 }
 
+size_t ZDICT_addEntropyTablesFromBuffer_advanced(void* dictBuffer, size_t dictContentSize, size_t dictBufferCapacity,
+                                                 const void* samplesBuffer, const size_t* samplesSizes, unsigned nbSamples,
+                                                 ZDICT_params_t params)
+{
+    unsigned compressionLevel = params.compressionLevel;
+    if (compressionLevel==0) compressionLevel = g_compressionLevel_default;
+
+    size_t hSize;
```
Comment (Cyan4973) [inline lib/dictBuilder/zdict.c]: Unfortunately, we only use variable declaration at beginning of scope, for maximum portability.
You can either declare `hSize` higher in the block,
or change `compressionLevel` declaration to become a one-liner : 

`unsigned const compressionLevel = params.compressionLevel ? params.compressionLevel : g_compressionLevel_default; `

which I would favor to grab the `const` property. 
But that's just a coding taste...

Comment (ot) [inline lib/dictBuilder/zdict.c]: Oh yeah sorry, my C++ habits :) I'll switch to the ternary operator.

**Comment by ot** [comment]
> The currently active feature branch is dev070.

Sure, I'll rebase on that branch.

**Comment by ot** [comment]
Looks like GitHub does not support changing the base branch of a PR, I'll open a new one.

**Comment by Cyan4973** [comment]
Indeed, I believe it's the number one feature request since a few years already ...

https://github.com/isaacs/github/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc

**Pull request closed without merging**